### PR TITLE
Fix rate limit of 600 requests per minutes on Gitlab.com

### PR DESCRIPTION
**Pull request by jdecaron**
This is a simple fix, maximum of 10 requests per seconds = maximum of 600 requests per minutes. It does slow down a bit the report but it prevent a fatal error. It's possible to design a more sophisticated fix but it will be more complex with more states to manage. It would be more prone to bugs. I favor simplicity over maximum speed.

### CODE REVIEW

**Comment by adescoms** [comment]
Hi, this PR would help us to solve https://github.com/kriskbx/gitlab-time-tracker/issues/82 and https://github.com/kriskbx/gitlab-time-tracker/issues/73 because we're overrating the GitLab limit (10 requests per second), can I help in something to move it forward? @kriskbx @jdecaron

File: src/models/base.js [hunk 1]
```diff
@@ -2,6 +2,9 @@ const request = require('request-promise-native');
 const url = require('url');
 const async = require('async');
 const crypto = require('crypto');
+const throttledQueue = require('throttled-queue');
+
+const throttle = throttledQueue(10, 1000);
```
Comment (cgdobre) [inline src/models/base.js:7]: Let's have a way to configure requests per second, as it is done now for parallel. Of course, this means also that:

- [ ]  the declaration of throttle needs to be moved inside the base object.

- [ ] the new throttle parameter needs to be added in the configuration

```suggestion
const throttledQueue = require('throttled-queue');
[...]
let throttle = throttledQueue(this._throttle, 1000);
[...]
```

File: src/models/base.js [hunk 2]
```diff
@@ -36,19 +39,21 @@ class base {
...
+                  proxy: this._proxy,
+                  resolveWithFullResponse: true,
+                  headers: {
+                      'PRIVATE-TOKEN': this.token
+                  }
+              }).then(response => {
+                  if (this.config.get('_createDump')) this.setDump(response, key);
+                  resolve(response);
+              }).catch(e => reject(e));
+            })
```
Comment (cgdobre) [inline src/models/base.js:56]: Let's move throttle() on line up to minimize the diff
```suggestion
return new Promise((resolve, reject) => throttle(() => {
            request.post(`${this.url}${path}`, {
                json: true,
                body: data,
                insecure: this._insecure,
                proxy: this._proxy,
                resolveWithFullResponse: true,
                headers: {
                    'PRIVATE-TOKEN': this.token
                }
            }).then(response => {
                if (this.config.get('_createDump')) this.setDump(response, key);
                resolve(response);
            }).catch(e => reject(e));
        }));
```

File: src/models/base.js [hunk 3]
```diff
@@ -67,18 +72,20 @@ class base {
...
```
Comment (cgdobre) [inline src/models/base.js:88]: same suggestion as above. Move throttle one line up and un-indent the request lines to minimize the diff.

**Comment by cgdobre** [comment]
PS: also update and include the yarn.lock file

**Comment by cgdobre** [comment]
PS2: goes without saying, if you need any help with the changes, just ask.

**Pull request merged**